mqtt/mqtt-broker/src/broker.rs

```diff
@@ -107,48 +121,44 @@ where
         BrokerState { retained, sessions }
     }
 
-    async fn process_message(
-        &mut self,
-        client_id: ClientId,
-        event: ClientEvent,
-    ) -> Result<(), Error> {
+    fn process_message(&mut self, client_id: ClientId, event: ClientEvent) -> Result<(), Error> {
         debug!("incoming: {:?}", event);
         let result = match event {
-            ClientEvent::ConnReq(connreq) => self.process_connect(client_id, connreq).await,
+            ClientEvent::ConnReq(connreq) => self.process_connect(client_id, connreq),
             ClientEvent::ConnAck(_) => {
                 info!("broker received CONNACK, ignoring");
                 Ok(())
             }
-            ClientEvent::Disconnect(_) => self.process_disconnect(client_id).await,
-            ClientEvent::DropConnection => self.process_drop_connection(client_id).await,
-            ClientEvent::CloseSession => self.process_close_session(client_id).await,
-            ClientEvent::PingReq(ping) => self.process_ping_req(client_id, ping).await,
+            ClientEvent::Disconnect(_) => self.process_disconnect(&client_id),
+            ClientEvent::DropConnection => self.process_drop_connection(&client_id),
+            ClientEvent::CloseSession => self.process_close_session(&client_id),
+            ClientEvent::PingReq(ref ping) => self.process_ping_req(&client_id, ping),
```

Comment:
```suggestion
            ClientEvent::PingReq(ping) => self.process_ping_req(&client_id, &ping),
```
Everywhere?

Reply:
You don't like the `ref`?

Reply:
Yeah, for some reason `ref` doesn't look as ergonomic as without `ref`. Feels like an additional mental burden.

Interesting, it's not only for me :)
https://rust-lang.github.io/rfcs/2005-match-ergonomics.html